lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/spatial/PointCloud.java

```diff
@@ -0,0 +1,286 @@
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * This class represents a point cloud based on the {@link Vector3d} coordinate system.
+ * <ul> <li>It consists of a reference point which represents the origin/center of this point cloud
+ * in absolute world coordinates.</li>
+ * <li>A rotation matrix used to translate relative end points of the point cloud to absolute coordinates.</li>
+ * <li>A list of points (x,y,z, hit type, distance to origin) forming this point cloud.</li>
+ * <li>Use one of the given {@link PointReference} formats to declare the reference frame of the given point lists. If
+ *     they are already in absolute world coordinates, use {@link PointReference#ABSOLUTE}. If the reference frame of the
+ *     points are {@link #origin} and  {@link #orientation}, then use {@link PointReference#RELATIVE}.
+ * </ul>
+ * When calling the methods {@link #getRelativeEndPoints()} or {@link #getAbsoluteEndPoints()}, a transformation of the stored points
+ * may be executed (and cached for re-use). E.g., if the point cloud is constructed using absolute coordinates
+ * (i.e. {@link PointReference#ABSOLUTE}), a call of {@link #getAbsoluteEndPoints()} will return the stored point list as-is, whereas
+ * a call of {@link #getRelativeEndPoints()} will transform the stored points to relative coordinates (relatively to  {@link #origin} and
+ * {@link #orientation}).
+ */
+public final class PointCloud implements Serializable {
+
+    private static final Predicate<Point> POINTS_ALL = p -> true;
+    private static final Predicate<Point> POINTS_WITH_HITS = Point::hasHit;
+
+    public enum PointReference {
+        ABSOLUTE(Transformation::absoluteToRelative, Transformation::noTransformation),
+        RELATIVE(Transformation::noTransformation, Transformation::relativeToAbsolute);
+
+        PointReference(Transformation toRelative, Transformation toAbsolute) {
+            this.toRelative = toRelative;
+            this.toAbsolute = toAbsolute;
+        }
+
+        private final Transformation toRelative;
+        private final Transformation toAbsolute;
+    }
+
+    private static final long serialVersionUID = 1L;
+
+    private final Vector3d origin;
+    private final RotationMatrix orientation;
+    private final long creationTime;
+
+    private final PointReference pointsReference;
+    private final List<Point> points;
+
+    private transient List<Point> absoluteEndPoints = null;
+    private transient List<Point> absoluteEndPointsWithHit = null;
+    private transient List<Point> relativeEndPoints = null;
+    private transient List<Point> relativeEndPointsWithHit = null;
+
+    /**
+     * Creates a new PointCloud based on a list of {@link Point}s. The points may be given (and then stored) either in absolute
+     * world coordinates, or relative coordinates compared to the given origin and orientation field. The reference frame  of the points
```

Comment:
There is a space too many here
```suggestion
     * world coordinates, or relative coordinates compared to the given origin and orientation field. The reference frame of the points
```